tests/test_submit.py

```diff
@@ -0,0 +1,264 @@
+import pathlib
+import subprocess
+from unittest import mock
+
+import click
+import pytest
+
+
+def git_command(
+    cwd: pathlib.Path, *args, capture_output=False
+) -> tuple[int, str | None]:
+    """Execute a git command in the given directory."""
+    result = subprocess.run(
+        ["git", *args],
+        cwd=cwd,
+        capture_output=capture_output,
+        text=True,
+    )
+    return result.returncode, result.stdout.strip() if capture_output else None
+
+
+def make_repo(repo_path: pathlib.Path) -> str | None:
+    """Creates a git repo with an initial commit and returns the branch name"""
+    repo_path.mkdir()
+    git_command(repo_path, "init")
+    git_command(repo_path, "config", "user.name", "Test User")
+    git_command(repo_path, "config", "user.email", "test@example.com")
+
+    # Create and commit an initial file
+    initial_file = repo_path / "initial.txt"
+    initial_file.write_text("initial content")
+
+    git_command(repo_path, "add", "initial.txt")
+    git_command(repo_path, "commit", "-m", "Initial commit")
+
+    # Get current branch name
+    returncode, branch = git_command(
+        repo_path, "branch", "--show-current", capture_output=True
+    )
+    assert returncode == 0
+    return branch
```

Comment:
Don't see the value of this function being separate from the fixture.

Reply:
it was useful 10 refactors back, and I forgot to fix it...